src/util/cmdline.cpp

```diff
@@ -145,6 +148,42 @@ const char *cmdlinet::getval(const char *option) const
   return value->second.front().c_str();
 }
 
+std::string cmdlinet::expand_path_string(std::string path) const
+{
+  return path;
+}
+
+// Will return empty string if invalid.
+std::string cmdlinet::get_config_file_location() const
+{
+  const auto envloc = std::getenv("ESBMC_CONFIG_FILE");
+  if (envloc)
+  {
+    const std::string config_path =
+      this->expand_path_string(std::string(envloc));
+    if (std::filesystem::exists(config_path))
+    {
+      return config_path;
+    }
+  }
+  else
+  {
+    // Load default config file if it exists.
+#ifdef WIN32
+#  define DEFAULT_CONFIG_PATH "%userprofile%\\esbmc.toml"
+#else
+#  define DEFAULT_CONFIG_PATH "~/.config/esbmc.toml"
+#endif
+    const std::string config_path =
+      this->expand_path_string(DEFAULT_CONFIG_PATH);
+    if (std::filesystem::exists(config_path))
+    {
+      return config_path;
+    }
+  }
+  return "";
```

Comment:
you can return an optional instead of an empty string.

Reply:
Done